First pass at rewriting photon emitters into a single cpp file

andrewgait opened the pull request:
Resolves #13

Making it possible to load photons either from a dict (via a VolumeGrid code object, or from a binary file) or XML (from a binary file).

Added a simple test script  to the SingleEmitter directory which can be used for any of the three cases.  (Note that it's not possible to load a code object from an XML file).

Leaving in draft form for now while I consider whether there are any improvements or simplifications to the code as it stands.

andrewgait committed:
First pass at rewriting photon emitters into a single cpp file reading either from a dict or XML

- Added a simple test script  to the SingleEmitter directory

andrewgait committed:
Add some documentation to top of photon emitter file

andrewgait committed:
Add some TODOs for possible required future updates

ianhinder commented on MPhys/Single_Emitter/single_emitter_test_new_single_test.py:
Probably good to turn this into a test in the test directory, so that when all the tests are run, we check that the new loading code gives the same result as before.

andrewgait replied:
This is covered by issue #4 so I will make a branch for that once this is merged.

andrewgait committed:
Simplify common code for photon loaders

andrewgait committed:
Update script to use matching CSV / binary files; add more detail on diff image

andrewgait committed:
Remove constant parameters that aren't used in the photon emitter

andrewgait committed:
Refer to correct emitter name in XML files

andrewgait committed:
Add basic tests for photon emitter

- Still need to test elements of photon emitter that are different from spot emitter

andrewgait commented:
I have now added some basic tests for the photon emitter in the same style as the tests for the spot emitter. I still need to add further tests to check that all methods for loading in data work, and also need to modify to test how the `sample_ray` and `sample_direction` functions behave in the photon emitter.

ianhinder commented:
Let's prioritise this, since it's almost finished.

andrewgait reviewed (commented):
This is good to merge now, and then follow up with further branches and PRs for issues #4 and #9.
